src/transformers/processing_utils.py

```diff
@@ -829,7 +829,12 @@ class MyProcessingKwargs(ProcessingKwargs, CommonKwargs, TextKwargs, ImagesKwarg
             for modality_key in ModelProcessorKwargs.__annotations__[modality].__annotations__.keys():
                 # init with tokenizer init kwargs if necessary
                 if modality_key in tokenizer_init_kwargs:
-                    default_kwargs[modality][modality_key] = tokenizer_init_kwargs[modality_key]
+                    value = (
+                        getattr(self.tokenizer, modality_key)
+                        if hasattr(self.tokenizer, modality_key)
+                        else tokenizer_init_kwargs[modality_key]
+                    )
+                    default_kwargs[modality][modality_key] = value
```

Comment:
I just realized the issue comes from this line when we use default tokenizer kwargs that were set during init. It a bit breaking if ppl change the attr like `padding_side` and that is not reflected in init_kwargs, so I think we can check first tokenizer attr and then fallback to the init kwargs
@molbap wdyt?

Reply:
Ah I see, yes that makes sense, it should be a "soft" check though to make sure we are not requiring anything at this stage, just that if it is present, it should be superseding. 
Handling priorities in this script is :zany_face: good catch though!

Reply:
yea, i think the check is soft as we first see if attribute if present, and if not fallback to the usual `init_kwargs[key]`